### PR TITLE
feat(sales/imports):externalised import of sales into catalog app

### PR DESCRIPTION
**Pull request by iPR0D1G1**
fixed: #6752

### CODE REVIEW

File: apps/catalog/catalog/src/app/dashboard/sales/import/import-sales.module.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    CommonModule,
+    FlexLayoutModule,
+    HttpClientModule,
+    RouterModule.forChild(routes),
+
+    // Material
+    MatIconModule,
+    MatButtonModule,
+
+    // Librairies
```
Comment (pldespaigne) [inline apps/catalog/catalog/src/app/dashboard/sales/import/import-sales.module.ts:43]: Libraries

File: apps/catalog/catalog/src/app/dashboard/sales/import/import-spreadsheet/import-spreadsheet.component.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import { AuthQuery } from '@blockframes/auth/+state';
+import { Subscription } from 'rxjs';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { allowedFiles } from '@blockframes/utils/utils';
+import { getMimeType } from '@blockframes/utils/file-sanitizer';
+
+export interface SpreadsheetImportEvent {
+  sheet: SheetTab,
+  fileType: string,
+}
```
Comment (pldespaigne) [inline apps/catalog/catalog/src/app/dashboard/sales/import/import-spreadsheet/import-spreadsheet.component.ts:15]: The same type is copy/pasted across 2 files, please define it once and import it when needed

File: apps/catalog/catalog/src/app/dashboard/sales/import/import-sales-container/import-sales-container.component.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  styleUrls: ['./import-sales-container.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ImportSalesContainerComponent implements OnInit {
+
+  @ViewChild('viewExtractedContractsComponent') viewExtractedContractsComponent: ViewExtractedContractsComponent;
+
+  public importedFiles = false;
+  public fileType: string;
+  public isUserBlockframesAdmin = false;
```
Comment (pldespaigne) [inline apps/catalog/catalog/src/app/dashboard/sales/import/import-sales-container/import-sales-container.component.ts:19]: this is not used

File: apps/catalog/catalog/src/app/dashboard/sales/import/import-sales-container/import-sales-container.component.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  selector: 'import-sales-container',
+  templateUrl: './import-sales-container.component.html',
+  styleUrls: ['./import-sales-container.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ImportSalesContainerComponent implements OnInit {
+
+  @ViewChild('viewExtractedContractsComponent') viewExtractedContractsComponent: ViewExtractedContractsComponent;
+
+  public importedFiles = false;
```
Comment (pldespaigne) [inline apps/catalog/catalog/src/app/dashboard/sales/import/import-sales-container/import-sales-container.component.ts:17]: this is not used

File: apps/catalog/catalog/src/app/dashboard/sales/import/import-spreadsheet/import-spreadsheet.component.ts [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ImportSpreadsheetComponent implements OnInit, OnDestroy {
+  @Output() importEvent = new EventEmitter<{ sheet: SheetTab, fileType: string }>();
+  public sheets: SheetTab[] = [];
+  public fileType = new FormControl('contracts');
+  public allowedTypes: string[] = [];
+  public mimeTypes: string[] = [];
+  private sub: Subscription;
+  private file: File;
```
Comment (pldespaigne) [inline apps/catalog/catalog/src/app/dashboard/sales/import/import-spreadsheet/import-spreadsheet.component.ts:30]: everything should be private except for `sheets` & `allowedTypes`

File: apps/catalog/catalog/src/app/dashboard/sales/import/import-spreadsheet/import-spreadsheet.component.ts [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    let sheetRange: string;
+    if (this.fileType.value === 'movies') {
+      sheetRange = 'A14:BZ1000';
+    } else if (this.fileType.value === 'contracts') {
+      sheetRange = 'A1:Q100';
+    } else if (this.fileType.value === 'organizations') {
+      sheetRange = 'A10:Z100';
+    } else {
+      sheetRange = 'A10:AD100';
+    }
```
Comment (pldespaigne) [inline apps/catalog/catalog/src/app/dashboard/sales/import/import-spreadsheet/import-spreadsheet.component.ts:94]: This look like duplicated code, I think it's better to re-use components than to copy/past them

**Review by pldespaigne — CHANGES_REQUESTED** [review]
I know the task wasn't well defined, that's my bad.
But please try to avoid copy pasting code from one place to another.
Please instead try to extract common logic in re-usable components/functions.

File: apps/catalog/catalog/src/app/dashboard/sales/import/import-sales-container/import-sales-container.component.ts (hunk 4 above)
Comment (iPR0D1G1) [reply apps/catalog/catalog/src/app/dashboard/sales/import/import-sales-container/import-sales-container.component.ts:17]: It is used within the corresponding template to either show the upload file prompt or a message indicating what happens after the file is processed.

File: apps/catalog/catalog/src/app/dashboard/sales/import/import-spreadsheet/import-spreadsheet.component.ts [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        const element = document.createElement('a');
+        element.setAttribute('href', url);
+        element.setAttribute('download', fileName);
+        const event = new MouseEvent('click');
+        element.dispatchEvent(event);
+      });
+  }
+
+
+}
```
Comment (pldespaigne) [inline apps/catalog/catalog/src/app/dashboard/sales/import/import-spreadsheet/import-spreadsheet.component.ts:105]: This is a duplicate from `libs\import\src\lib\components\import-spreadsheet\import-spreadsheet.component.ts`

**Review by pldespaigne — CHANGES_REQUESTED** [review]

**Pull request closed without merging**